### PR TITLE
Allow percent % in path and file names #561

### PR DESCRIPTION
**Pull request by moldover**

### CODE REVIEW

File: src/main/java/network/brightspots/rcv/Logger.java [hunk 1]
```diff
@@ -116,19 +118,24 @@ static void removeTabulationFileLogging() {
...
-    logger.log(Level.WARNING, String.format(message, obj));
+    log(Level.WARNING, message, obj);
   }
 
   static void severe(String message, Object... obj) {
-    logger.log(Level.SEVERE, String.format(message, obj));
+    log(Level.SEVERE, message, obj);
+  }
+
+  static void log(Level level, String message, Object... obj) {
```
Comment (HEdingfield) [inline src/main/java/network/brightspots/rcv/Logger.java:136]: Recommend keeping this `private` until it's actually needed outside this class.

File: src/main/java/network/brightspots/rcv/Logger.java [hunk 2]
```diff
@@ -99,9 +99,11 @@ static void setup() {
 
   // adds file logging for a tabulation run
   static void addTabulationFileLogging(String outputPath) throws IOException {
-    // use Level.FINE to capture audit info
-    tabulationHandler =
-        new FileHandler(outputPath, LOG_FILE_MAX_SIZE_BYTES, TABULATION_LOG_FILE_COUNT, true);
+    // replace any instances of % with %% because the FileHandler pattern generator requires this
+    tabulationHandler = new FileHandler(outputPath.replace("%", "%%"),
```
Comment (HEdingfield) [inline src/main/java/network/brightspots/rcv/Logger.java:103]: Would this work as intended if a file name has a double percentage in the name?

**Review by HEdingfield — COMMENTED** [review]
Were you able to test this with the files included in [this issue](https://github.com/BrightSpots/rcv/issues/564#issue-860701500)?

File: src/main/java/network/brightspots/rcv/Logger.java (hunk 2 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/Logger.java:103]: Yep, I tried this with multiple percent characters and that part works, but as I'm checking again I realize I broke the %g functionality here so it's not quite correct yet.  Will push a fix for that.

Comment (moldover) [reply src/main/java/network/brightspots/rcv/Logger.java]: Just pushed a fix for this bit.  Go ahead and get hog-wild with your path names.

**Comment by moldover** [comment]
@HEdingfield + @andyanderson yep - was able to tabulate:

2022-06-06 18:53:47 PDT INFO: Round: 4
2022-06-06 18:53:47 PDT INFO: Winning threshold set to 14.
2022-06-06 18:53:47 PDT INFO: Candidate "Strawberry" got 15 vote(s).
2022-06-06 18:53:47 PDT INFO: Candidate "Grapefruit" got 11 vote(s).
2022-06-06 18:53:47 PDT INFO: Candidate "Strawberry" was elected in round 4 with 15 votes.

Congratulations to Strawberry.

File: src/main/java/network/brightspots/rcv/Logger.java [hunk 3]
```diff
@@ -98,14 +98,24 @@ static void setup() {
...
+
+    tabulationHandler = new FileHandler(tabulationLogPattern,
+            LOG_FILE_MAX_SIZE_BYTES,
+            TABULATION_LOG_FILE_COUNT,
+            true);
     tabulationHandler.setFormatter(formatter);
     tabulationHandler.setLevel(Level.FINE);
     logger.addHandler(tabulationHandler);
-    info("Tabulation logging to: %s", outputPath.replace("%g", "*"));
+    info("Tabulation logging to: %s", tabulationLogPattern.replace("%g", "0"));
```
Comment (HEdingfield) [inline src/main/java/network/brightspots/rcv/Logger.java:118]: Why this change? It seems like it makes it look like it's always pointing to log index 0 when that may not actually be the case.

**Review by HEdingfield — COMMENTED** [review]
Big up the strawberry.

File: src/main/java/network/brightspots/rcv/Logger.java (hunk 3 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/Logger.java:118]: I believe this is an improvement over the current message.  At the time it's printed it's accurate and it is less confusing than the current asterisk message, which might mean something to folks acquainted with the notion of wildcards, but not to our users.

Comment (HEdingfield) [reply src/main/java/network/brightspots/rcv/Logger.java:118]: I guess this begs the question of why we even include the log index at all if we know it's always going to be 0? I assume it serves some purpose and, if so, I think we should keep this as-is. We've already got an established precedent of using * in a log name for the execution log message, and I think it's a pretty well-understood concept.

In any case, this PR looks good, thanks for fixing this!

**Review by HEdingfield — APPROVED** [review]
